client/src/apis/fetcher.ts

```diff
@@ -1,120 +1,165 @@
+  await request({method: 'POST', headers, ...args});
 };
 
-export const requestPostWithResponse = async <T>({headers = {}, ...args}: RequestProps): Promise<T> => {
-  const response = await fetcher({method: 'POST', headers, ...args});
+export const requestPostWithResponse = async <T>({headers = {}, ...args}: RequestMethodProps): Promise<T> => {
+  const response = await request({method: 'POST', headers, ...args});
 
   const data: T = await response!.json();
   return data;
 };
 
-export const requestDelete = ({headers = {}, ...args}: RequestProps) => {
-  return fetcher({method: 'DELETE', headers, ...args});
+export const requestDelete = ({headers = {}, ...args}: RequestMethodProps) => {
+  return request({method: 'DELETE', headers, ...args});
 };
 
-const fetcher = ({baseUrl = API_BASE_URL, method, endpoint, headers, body, queryParams}: FetcherProps) => {
-  const options = {
-    method,
+const prepareRequest = ({baseUrl = API_BASE_URL, method, endpoint, headers, body, queryParams}: RequestProps) => {
+  let url = `${baseUrl}${endpoint}`;
+
+  if (queryParams) url += `?${objectToQueryString(queryParams)}`;
+
+  const requestInit: RequestInitWithMethod = {
     credentials: 'include',
     headers: {
       'Content-Type': 'application/json',
       ...headers,
     },
+    method,
     body: body ? JSON.stringify(body) : null,
   };
 
-  let url = `${baseUrl}${endpoint}`;
-
-  if (queryParams) url += `?${objectToQueryString(queryParams)}`;
-
-  return errorHandler({url, options, body: JSON.stringify(body)});
+  return {url, requestInit};
 };
 
-const errorHandler = async ({url, options, body}: ErrorHandlerProps) => {
+const executeRequest = async ({url, requestInit, errorHandlingStrategy}: WithErrorHandlingStrategy<FetchType>) => {
   try {
-    const response: Response = await fetch(url, options);
+    const response: Response = await fetch(url, requestInit);
 
     if (!response.ok) {
-      const serverErrorInfo: ErrorInfo = await response.json();
-
-      throw new FetchError({
-        status: response.status,
-        requestBody: body,
-        endpoint: response.url,
-        errorInfo: serverErrorInfo,
-        name: serverErrorInfo.errorCode,
-        message: serverErrorInfo.message || '',
-        method: options.method,
+      throw await createError({
+        response,
+        body: requestInit.body ? JSON.stringify(requestInit.body) : null,
+        requestInit,
+        errorHandlingStrategy,
       });
     }
 
     return response;
   } catch (error) {
     if (error instanceof Error) {
-      throw error; // 그대로 FetchError || Error 인스턴스를 던집니다.
+      throw error; // 그대로 RequestError 또는 Error 인스턴스를 던집니다.
     }
 
-    throw new Error(UNKNOWN_ERROR);
+    throw error;
   }
 };
+
+const request = async (props: WithErrorHandlingStrategy<RequestProps>) => {
+  const {url, requestInit} = prepareRequest(props);
+  return executeRequest({url, requestInit, errorHandlingStrategy: props.errorHandlingStrategy});
+};
```

Comment:
보낼 준비를 먼저 하고, 실제 요청을 보낸다!
더 보기 좋아진 것 같아요😊😊

그리고 prepareRequest에는 errorHandlingStrategy가 필요없으니깐 이렇게 구조분해 할당해도 될 것 같아요!

```suggestion
const request = async ({errorHandlingStrategy, ...props}: WithErrorHandlingStrategy<RequestProps>) => {
  const {url, requestInit} = prepareRequest(props);
  return executeRequest({url, requestInit, errorHandlingStrategy});
};
```

Reply:
변수명이 안그래도 너무 길어서 거슬렸는데 확실히 더 읽기 좋네요...! 에러 핸들링 기능에서 사용되는 변수명들을 정리하려고 했는데 그때 이것도 반영하겠습니다. 👍